components/director/internal/domain/bundleinstanceauth/service_test.go

```diff
@@ -926,6 +1116,524 @@ func TestService_RequestDeletion(t *testing.T) {
+		},
+		{
+			Name: "Success when not adding any new scenario",
+			LabelSvcFn: func() *automock.LabelService {
+				return &automock.LabelService{}
+			},
+			ScenarioSvcFn: func() *automock.ScenarioService {
+				return &automock.ScenarioService{}
+			},
+			ExistingScenariosParam: []string{scenarioToKeep},
+			InputScenariosParam:    []string{scenarioToKeep},
+			ExpectedError:          nil,
+		},
+		{
+			Name: "Success when both remove some old scenarios and add new ones",
+			LabelSvcFn: func() *automock.LabelService {
+				svc := &automock.LabelService{}
+				svc.On("UpsertLabel", contextThatHasTenant(testTenant), testTenant, matchScenarios([]string{scenarioToKeep, scenarioToAdd1})).
+					Return(nil).Once()
+				svc.On("UpsertLabel", contextThatHasTenant(testTenant), testTenant, matchScenarios([]string{scenarioToKeep, scenarioToAdd2})).
+					Return(nil).Once()
+				return svc
+			},
+			ScenarioSvcFn: func() *automock.ScenarioService {
+				svc := &automock.ScenarioService{}
+				svc.On("GetRuntimeScenarioLabelsForAnyMatchingScenario", contextThatHasTenant(testTenant), []string{scenarioToKeep}).
+					Return(existingRuntimeScenarioLabels, nil).Once()
+
+				rmt1Labels := []model.Label{{ObjectID: "runtime-1", Key: model.ScenariosKey, Value: []string{scenarioToKeep, scenarioToAdd1}}}
+				rmt2Labels := []model.Label{{ObjectID: "runtime-2", Key: model.ScenariosKey, Value: []string{scenarioToKeep, scenarioToAdd2}}}
+
+				svc.On("GetBundleInstanceAuthsScenarioLabels", contextThatHasTenant(testTenant), applicationID, "runtime-1").Return(rmt1Labels, nil).Once()
+				svc.On("GetBundleInstanceAuthsScenarioLabels", contextThatHasTenant(testTenant), applicationID, "runtime-2").Return(rmt2Labels, nil).Once()
+				return svc
+			},
+			ExistingScenariosParam: []string{scenarioToRemove, scenarioToKeep},
+			InputScenariosParam:    []string{scenarioToKeep, scenarioToAdd1, scenarioToAdd2},
+			ExpectedError:          nil,
+		},
+		{
+			Name: "Success when there are no scenario to keep",
+			LabelSvcFn: func() *automock.LabelService {
+				return &automock.LabelService{}
+			},
+			ScenarioSvcFn: func() *automock.ScenarioService {
+				return &automock.ScenarioService{}
+			},
+			ExistingScenariosParam: []string{scenarioToRemove},
+			InputScenariosParam:    []string{scenarioToAdd1},
+			ExpectedError:          nil,
+		},
+		{
+			Name: "Success when there are NO runtimes that application is connected",
+			LabelSvcFn: func() *automock.LabelService {
+				return &automock.LabelService{}
+			},
+			ScenarioSvcFn: func() *automock.ScenarioService {
+				svc := &automock.ScenarioService{}
+				svc.On("GetRuntimeScenarioLabelsForAnyMatchingScenario", contextThatHasTenant(testTenant), []string{scenarioToKeep}).Return([]model.Label{}, nil).Once()
+				return svc
+			},
+			ExistingScenariosParam: []string{scenarioToKeep},
+			InputScenariosParam:    []string{scenarioToKeep, scenarioToAdd1},
+			ExpectedError:          nil,
+		},
```

Comment:
Since it uses `lister` as a backend such issue won't happen. But it will return `nil` instead of empty array. Maybe this can be point for fix the test only